js/app/interfaces/card.js

```diff
@@ -302,18 +299,18 @@ const Card = new Lang.Interface({
         if (!this.model.thumbnail_uri)
             return;
 
-        let coveredFrame = new ImageCoverFrame.ImageCoverFrame();
-        frame.add(coveredFrame);
-
-        let cancellable = null;
-        let file = Gio.File.new_for_uri(this.model.thumbnail_uri);
-        let stream = file.read(cancellable);
-        coveredFrame.set_content(stream);
-        frame.visible = true;
+        let bg_provider = new Gtk.CssProvider();
+        bg_provider.load_from_data('frame {\
+            background-size: cover;\
```

Comment:
This might not actually be more performant, remember holding off on this change a while ago for that reason. Gtk won't attempt to cache anything, it will redraw (and scale) the background every time. Which for a big background image getting sized down could be slow. Template B is a good example.

However if this is going to land in 3.1 along with some gsk goodies. Suspect this is a non issue.